x/profile/keeper/msg_server_update_coordinator_description_test.go

```diff
@@ -0,0 +1,95 @@
+package keeper
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/stretchr/testify/require"
+	"github.com/tendermint/spn/testutil/sample"
+	"github.com/tendermint/spn/x/profile/types"
+)
+
+func TestMsgUpdateCoordinatorDescription(t *testing.T) {
+	var (
+		addr        = sample.AccAddress()
+		msgCoord    = msgCreateCoordinator()
+		ctx, k, srv = setupMsgServerAndKeeper(t)
+		wCtx        = sdk.WrapSDKContext(ctx)
+	)
+	if _, err := srv.CreateCoordinator(wCtx, &msgCoord); err != nil {
+		t.Fatal(err)
+	}
+	tests := []struct {
+		name string
+		msg  types.MsgUpdateCoordinatorDescription
+		err  error
+	}{
+		{
+			name: "not found address",
+			msg: types.MsgUpdateCoordinatorDescription{
+				Address: addr,
+			},
+			err: sdkerrors.Wrap(types.ErrCoordAddressNotFound, addr),
+		}, {
+			name: "update one value",
+			msg: types.MsgUpdateCoordinatorDescription{
+				Address: msgCoord.Address,
+				Description: &types.CoordinatorDescription{
+					Identity: "update",
+				},
+			},
+		}, {
+			name: "update all values",
+			msg: types.MsgUpdateCoordinatorDescription{
+				Address: msgCoord.Address,
+				Description: &types.CoordinatorDescription{
+					Identity: "update",
+					Website:  "update",
+					Details:  "update",
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var oldCoord types.Coordinator
+			if tt.err == nil {
+				coordByAddr, found := k.GetCoordinatorByAddress(ctx, tt.msg.Address)
+				require.True(t, found, "coordinator by address not found")
+				oldCoord = k.GetCoordinator(ctx, coordByAddr.CoordinatorId)
+			}
+
+			_, err := srv.UpdateCoordinatorDescription(wCtx, &tt.msg)
+			if tt.err != nil {
+				require.ErrorIs(t, err, tt.err)
+				return
+			}
+			require.NoError(t, err)
+
+			coordByAddr, found := k.GetCoordinatorByAddress(ctx, tt.msg.Address)
+			coord := k.GetCoordinator(ctx, coordByAddr.CoordinatorId)
+			require.True(t, found, "coordinator id not found")
```

Comment:
It would be cleaner to have it above `GetCoordinator` otherwise we call `GetCoordinator` with `CoordinatorId` 0 but this is not blocking because the test will fail anyway